### PR TITLE
dependencies: updating to `v0.20230725.1205316` of `github.com/hashicorp/go-azure-sdk`

### PR DESCRIPTION
**Pull request by wuxu92**
required by #22400

### CODE REVIEW

File: internal/clients/client.go [hunk 1]
```diff
@@ -331,7 +331,9 @@ func (client *Client) Build(ctx context.Context, o *common.ClientOptions) error
 	if client.ConfidentialLedger, err = confidentialledger.NewClient(o); err != nil {
 		return fmt.Errorf("building clients for ConfidentialLedger: %+v", err)
 	}
-	client.Connections = connections.NewClient(o)
+	if client.Connections, err = connections.NewClient(o); err != nil {
+		return fmt.Errorf("building clinets for Connections: %+v", err)
```
Comment (stephybun) [inline internal/clients/client.go:335]: This is misspelled everywhere, please fix these 
```suggestion
		return fmt.Errorf("building clients for Connections: %+v", err)
```

File: internal/services/connections/client/client.go [hunk 2]
```diff
@@ -14,15 +16,21 @@ type Client struct {
 	ManagedApisClient *managedapis.ManagedAPIsClient
 }
 
-func NewClient(o *common.ClientOptions) *Client {
-	connectionsClient := connections.NewConnectionsClientWithBaseURI(o.ResourceManagerEndpoint)
-	o.ConfigureClient(&connectionsClient.Client, o.ResourceManagerAuthorizer)
+func NewClient(o *common.ClientOptions) (*Client, error) {
+	connectionsClient, err := connections.NewConnectionsClientWithBaseURI(o.Environment.ResourceManager)
+	if err != nil {
+		return nil, fmt.Errorf("build connections Client: %+v", err)
```
Comment (stephybun) [inline internal/services/connections/client/client.go:22]: Can we be consistent with the error messages and casing
```suggestion
		return nil, fmt.Errorf("building Connections client: %+v", err)
```

File: internal/services/connections/client/client.go [hunk 3]
```diff
@@ -14,15 +16,21 @@ type Client struct {
...
+	}
+	o.Configure(connectionsClient.Client, o.Authorizers.ResourceManager)
 
-	managedApisClient := managedapis.NewManagedAPIsClientWithBaseURI(o.ResourceManagerEndpoint)
-	o.ConfigureClient(&managedApisClient.Client, o.ResourceManagerAuthorizer)
+	managedApisClient, err := managedapis.NewManagedAPIsClientWithBaseURI(o.Environment.ResourceManager)
+	if err != nil {
+		return nil, fmt.Errorf("build connections managedApi Client: %+v", err)
```
Comment (stephybun) [inline internal/services/connections/client/client.go:28]:
```suggestion
		return nil, fmt.Errorf("building Managed Api client: %+v", err)
```

File: internal/services/mysql/client/client.go [hunk 4]
```diff
@@ -29,13 +29,19 @@ type Client struct {
...
 
-func NewClient(o *common.ClientOptions) *Client {
-	flexibleServersMetaClient := flexibleServers_v2021_05_01.NewClientWithBaseURI(o.ResourceManagerEndpoint, func(c *autorest.Client) {
-		o.ConfigureClient(c, o.ResourceManagerAuthorizer)
+func NewClient(o *common.ClientOptions) (*Client, error) {
+	flexibleServersMetaClient, err := flexibleServers_v2021_05_01.NewClientWithBaseURI(o.Environment.ResourceManager, func(c *resourcemanager.Client) {
+		o.Configure(c, o.Authorizers.ResourceManager)
 	})
+	if err != nil {
+		return nil, fmt.Errorf("build flexible servers meta Client: %+v", err)
```
Comment (stephybun) [inline internal/services/mysql/client/client.go:37]:
```suggestion
		return nil, fmt.Errorf("building Flexible Servers client: %+v", err)
```

File: internal/services/mysql/client/client.go [hunk 5]
```diff
@@ -29,13 +29,19 @@ type Client struct {
...
+	}
 
-	azureADAdministratorsClient := azureadadministrators.NewAzureADAdministratorsClientWithBaseURI(o.ResourceManagerEndpoint)
-	o.ConfigureClient(&azureADAdministratorsClient.Client, o.ResourceManagerAuthorizer)
+	azureADAdministratorsClient, err := azureadadministrators.NewAzureADAdministratorsClientWithBaseURI(o.Environment.ResourceManager)
+	if err != nil {
+		return nil, fmt.Errorf("build azure ADAdministrators Client: %+v", err)
```
Comment (stephybun) [inline internal/services/mysql/client/client.go:42]:
```suggestion
		return nil, fmt.Errorf("building Azure AD Administrators client: %+v", err)
```

File: internal/services/orbital/client/client.go [hunk 6]
```diff
@@ -18,19 +20,28 @@ type Client struct {
...
+	if err != nil {
+		return nil, fmt.Errorf("build orbital contact profile Client: %+v", err)
+	}
+	o.Configure(contactProfileClient.Client, o.Authorizers.ResourceManager)
 
-	contactClient := contact.NewContactClientWithBaseURI(o.ResourceManagerEndpoint)
-	o.ConfigureClient(&contactClient.Client, o.ResourceManagerAuthorizer)
+	contactClient, err := contact.NewContactClientWithBaseURI(o.Environment.ResourceManager)
+	if err != nil {
+		return nil, fmt.Errorf("build orbital contact Client: %+v", err)
```
Comment (stephybun) [inline internal/services/orbital/client/client.go:38]: Please fix the error messages here with consistent casing and grammar `building ...`

**Review by stephybun — DISMISSED** [review]

File: internal/clients/client.go (hunk 1 above)
Comment (wuxu92) [reply internal/clients/client.go]: sorry it's a typo when recording the macro. updated now.

**Review by tombuildsstuff — APPROVED** [review]

**Comment by tombuildsstuff** [comment]
Pushed a commit to update the Check function for mysql to include a context, but this otherwise LGTM 👍

**Pull request merged**